library/protocol.cpp

```diff
@@ -5,513 +5,604 @@
 #include "rapidjson/document.h"
 
 namespace msgpack {
+MSGPACK_API_VERSION_NAMESPACE(MSGPACK_DEFAULT_API_NS) {
+namespace adaptor {
 using namespace ioremap::elliptics;
 
-inline dnet_time &operator >>(msgpack::object o, dnet_time &v) {
-	if (o.type != msgpack::type::ARRAY || o.via.array.size != 2) {
-		throw msgpack::type_error();
-	}
-
-	object *p = o.via.array.ptr;
-	p[0].convert(&v.tsec);
-	p[1].convert(&v.tnsec);
-	return v;
-}
-
-template <typename Stream>
-inline msgpack::packer<Stream> &operator <<(msgpack::packer<Stream> &o, const dnet_time &v) {
-	o.pack_array(2);
-	o.pack_fix_uint64(v.tsec);
-	o.pack_fix_uint64(v.tnsec);
-	return o;
-}
+template<>
+struct convert<dnet_time> {
```

Comment:
JFMI: Am I understand correctly: it's a convient way to create custom convertors, and this opportunity is introduced in `v2.1.5-1`? In previous code used stream pattern, but now used conversion "method".

Reply:
Yes, but it was introduced in v1.0 or v.1.1 (https://github.com/msgpack/msgpack-c/wiki/v1_1_cpp_adaptor#non-intrusive-approach).